utils/wasm-builder/src/stack_end.rs

```diff
@@ -99,11 +81,323 @@ pub fn insert_stack_end_export(
     }
 }
 
-#[test]
-fn assembly_script_stack_pointer() {
-    use pwasm_utils::parity_wasm::elements;
+/// If `_start` export function exists, then insert this function call in the beginning of
+/// each export function.
+///
+/// If `_start` function does not exist, then do nothing, and returns Ok.
+/// If `_start` export exists, but by some reason we cannot insert its call in export functions,
+/// then returns Error.
+pub fn insert_start_call_in_export_funcs(module: &mut Module) -> Result<(), &'static str> {
+    let start_func_index = if let Some(start) = module
+        .export_section()
+        .ok_or("Cannot find export section")?
+        .entries()
+        .iter()
+        .find(|export| export.field() == "_start")
+    {
+        match start.internal() {
+            Internal::Function(index) => *index,
+            _ => return Err("_start export is not a function"),
+        }
+    } else {
+        return Ok(());
+    };
+
+    for export_name in optimize::FUNC_EXPORTS {
+        let Some(export) = module
+            .export_section()
+            .ok_or("Cannot find export section")?
+            .entries()
```

Comment:
Do we need to call this chain every time inside a loop?

Reply:
Yes, because `module` is used as mut and const inside loop. And this check is not expensive.